python/ray/tune/tests/test_tune_restore.py

```diff
@@ -151,6 +152,29 @@ def testExperimentInterrupted(self):
 
         shutil.rmtree(local_dir)
 
+    def testInterruptDisabledInWorkerThread(self):
+        # https://github.com/ray-project/ray/issues/22295
+        # This test will hang without the proper patch because tune.run will fail.
+
+        event = threading.Event()
```

Comment:
Do we need `event` to coordinate?
How about just `thread.join()`. I thought that would just wait for the thread to finish automatically - the same thing as what `event` is trying to do.

Reply:
Ah the issue here if the thread has exception, thread.join() will still succeed without re-raising anything.